samples/debugmalloc/README.md

```diff
@@ -0,0 +1,328 @@
+# The debug malloc sample
+
+- Introduction about debug malloc.
+- Demonstrate how to enable debug malloc through compiling options.
+- explanation and sample of debug malloc public apis.
+
+## Introduction
+
+Debug malloc is a tool that helps identifying memory leaks in enclaves. The
+main logic of debug malloc is as follows.
+
+- Whenever memory is allocated, debug malloc captures the callstack and
+associates it with the allocated memory. Additionally, debug malloc sets the
+contents of the allocated memory to zero.
+
+- Whenever memory is freed, it clears the contents to a fixed pattern to help
+detect access of freed memory. It also frees the associated callstack
+information.
+
+- When an enclave terminates, debug malloc prints information about leaks---
+those objects that have been allocated, but not freed.
+
+According to the user's developing experience, there are two major problems
```

Comment:
```suggestion
There are two major problems
```

(We don't refer to a "developer" as a "user", a user is who runs the programs the developer develops.  However here this phrase isn't important and can simply be omitted without loss of meaning.)